service/frontend/adminHandler.go

```diff
@@ -421,6 +423,237 @@ func (adh *AdminHandler) GetWorkflowExecutionRawHistory(
 	return result, nil
 }
 
+// GetWorkflowExecutionRawHistoryV2 - retrieves the history of workflow execution
+func (adh *AdminHandler) GetWorkflowExecutionRawHistoryV2(
+	ctx context.Context,
+	request *admin.GetWorkflowExecutionRawHistoryV2Request,
+) (resp *admin.GetWorkflowExecutionRawHistoryV2Response, retError error) {
+
+	defer log.CapturePanic(adh.GetLogger(), &retError)
+	scope := metrics.AdminGetWorkflowExecutionRawHistoryV2Scope
+	sw := adh.startRequestProfile(scope)
+	defer sw.Stop()
+
+	if err := adh.validateGetWorkflowExecutionRawHistoryV2Request(
+		request,
+	); err != nil {
+		return nil, adh.error(err, scope)
+	}
+	domainID, err := adh.domainCache.GetDomainID(request.GetDomain())
+	if err != nil {
+		return nil, adh.error(err, scope)
+	}
+
+	execution := request.Execution
+	response, err := adh.history.GetMutableState(ctx, &h.GetMutableStateRequest{
+		DomainUUID: common.StringPtr(domainID),
+		Execution:  execution,
+	})
+	if err != nil {
+		return nil, adh.error(err, scope)
+	}
+
+	versionHistories := persistence.NewVersionHistoriesFromThrift(
+		response.GetVersionHistories(),
+	)
+	startEventID, endEventID, versionHistory, err := adh.getEventRange(
+		request,
+		versionHistories,
+	)
+	if err != nil {
+		return nil, adh.error(err, scope)
+	}
+
+	token, err := adh.preparePaginationToken(
+		ctx,
+		request,
+		versionHistory.GetBranchToken(),
+		startEventID,
+		endEventID)
+	if err != nil {
+		return nil, adh.error(err, scope)
+	}
+
+	pageSize := int(request.GetMaximumPageSize())
+	shardID := common.WorkflowIDToHistoryShard(
+		execution.GetWorkflowId(),
+		adh.numberOfHistoryShards,
+	)
+	rawHistoryResponse, err := adh.historyV2Mgr.ReadRawHistoryBranch(&persistence.ReadHistoryBranchRequest{
+		BranchToken:   token.BranchToken,
+		MinEventID:    startEventID,
```

Comment:
```go
MinEventID:    startEventID + 1,
```

Reply:
the function `getEventRange` returns a inclusive / inclusive range while the range to return is exclusive / exclusive

`ReadRawHistoryBranch` take in a inclusive / exclusive range, so the `+1`

Reply:
updated